### PR TITLE
Fix date display bugs in project table and activity log

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/5732
https://github.com/cityofaustin/atd-data-tech/issues/9169
https://github.com/cityofaustin/atd-data-tech/issues/7121

This PR adds a couple helpers for formatting date type and time stamp with time zone data type coming from the database, and it replaces date and time formatting in various files with those helpers. I'm super open to feedback on the function names since it was hard to describe some of the formats - like how I called "January 21, 2022" format "US expanded" (came from [this wiki article](https://en.wikipedia.org/wiki/Date_and_time_notation_in_the_United_States) 🤷).

## Testing
Devs - you can run the unit tests from the command line using `npm run test`

**URL to test:** 
https://deploy-preview-653--atd-moped-main.netlify.app/moped/projects/

**Steps to test:**
Create a new project (or go to an existing one) and add a Construction phase with a start date and a Complete phase with an end date in the Timeline tab.

Go to the [project table](https://deploy-preview-653--atd-moped-main.netlify.app/moped/projects) and look at the Construction Start and Completion Date column values. They should match the dates that you entered in the Timeline tab.
Screen Shot 2022-05-20 at 11 38 36 AM

Go back to the project that you added the phases to and go to the Activity Log tab. You should see the activity entries for when you added the phases. The date in the log next to each entry should match the date the update was made. Then, click on the Details of one of the activity entries, and you should see the accurate date and time for the update that you made.
Screen Shot 2022-05-20 at 11 37 35 AM

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [x] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectActivityLog.js [hunk 1]
```diff
@@ -232,7 +225,7 @@ const ProjectActivityLog = () => {
                         width="5%"
                         className={classes.tableCell}
                       >
-                        {formatDate(change.created_at)}
+                        {formatTimeStampZType(change.created_at)}
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectActivityLog.js:228]: Fixes date shown in Activity Log https://github.com/cityofaustin/atd-data-tech/issues/7121

File: moped-editor/src/views/projects/projectView/ProjectActivityLogDialog.js [hunk 2]
```diff
@@ -107,10 +111,12 @@ const ProjectActivityLogDialog = ({ activity_id, handleClose }) => {
   };
 
   const getDateTime = data => {
-    const date = data?.moped_activity_log[0]?.created_at;
+    const timestampz = data?.moped_activity_log[0]?.created_at;
 
     try {
-      return new Date(date).toLocaleString('en-US', {timeZone: 'UTC'});
+      return `${formatTimeStampZType(timestampz)}, ${makeFullTimeFromTimeStampZ(
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectActivityLogDialog.js:117]: Fixes time stamp in Activity Log dialog (details view) https://github.com/cityofaustin/atd-data-tech/issues/7121

File: moped-editor/src/views/projects/projectsListView/ProjectsListViewTable.js [hunk 3]
```diff
@@ -420,16 +421,14 @@ const ProjectsListViewTable = ({ title, query, searchTerm, referenceData }) => {
...
+      render: entry => formatDateType(entry.construction_start_date),
     },
     {
       title: "Project completion",
       field: "completion_end_date",
       hidden: hiddenColumns["completion_end_date"],
       emptyValue: "-",
-      render: entry =>
-        new Date(entry.completion_end_date).toLocaleDateString("en-US"),
+      render: entry => formatDateType(entry.completion_end_date),
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectsListView/ProjectsListViewTable.js:431]: https://github.com/cityofaustin/atd-data-tech/issues/9169

**Review by mateoclarke — APPROVED** [review]
Looks great! super cool that `npm run test` just works! 🃏 

Screen Shot 2022-05-20 at 5 27 04 PM

File: moped-editor/src/utils/dateAndTime.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  });
+};
+
+/**
+ * Format timestampz string from the DB ("2022-05-17T22:37:26.072259+00:00")
+ * into human-readable locale date in MM/DD/YYYY format
+ * @param {String} timeStampZString - Timestampz string in ISO format
+ * @return {String} Date formatted as MM/DD/YYYY
+ */
+export const formatTimeStampZType = timeStampZString => {
```
Comment (mateoclarke) [inline moped-editor/src/utils/dateAndTime.js:20]: should this be `formatTimeStampTZType`? Seems consistent throughout at least.

Screen Shot 2022-05-20 at 5 31 13 PM

Comment (mddilley) [reply moped-editor/src/utils/dateAndTime.js:20]: Good call, yes, I will update this naming with TZ. I also added the run test script command to the PR description.

**Review by chiaberry — APPROVED** [review]

**Review by frankhereford — APPROVED** [review]
I'm in agreement with @mateoclarke: the addition of using tests in a test suite is just flat out cool and inspiring. Including tests sounds like just a "how could it be bad" thing for us to do when it's appropriate. I'm looking forward to learning how to write good, useful ones.

I think this PR is ready to ship. 🚢 👍 🙏 

As an aside, I this work inspired me to look more broadly at how we handle time data, and I have created [a new issue for us to discuss](https://github.com/cityofaustin/atd-data-tech/issues/9240).

File: moped-editor/src/utils/test/dateAndTime.test.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import {
+  formatDateType,
+  formatTimeStampZType,
+  makeUSExpandedFormDateFromTimeStampZ,
+  makeHourAndMinutesFromTimeStampZ,
+  makeFullTimeFromTimeStampZ,
+} from "../dateAndTime";
+
+describe("formatDateType()", () => {
+  it("formats PostgreSQL date type as MM/DD/YYYY", () => {
```
Comment (johnclary) [inline moped-editor/src/utils/test/dateAndTime.test.js:10]: this API is fun!

**Comment by johnclary** [comment]
super cool @mddilley. i'm new to js testing. you have any preferred docs for this test suite?

**Comment by mateoclarke** [comment]
> super cool @mddilley. i'm new to js testing. you have any preferred docs for this test suite?

Mike shared this last week: https://github.com/react-boilerplate/react-boilerplate/blob/master/docs/testing/unit-testing.md

**Comment by johnclary** [comment]
Excellent - thanks @mateoclarke

**Review by amenity — APPROVED** [review]
Beautiful, thanks @mddilley!

**Pull request merged**